# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webscheduling/WebTaskSchedulerMainThread.h
# Commit: 4739af17e757
# Full Hash: 4739af17e757724c0833a8564579833249389566
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-02-27 21:52:52
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Allow high priority tasks to run before timers r=smaug
#   
#   In Gecko, unlike other runnables where the TaskController
#   picks one each at a time, the TimeoutManager allows multiple
#   timers to run sequentially without allowing other runnables
# ==============================================================================

diff -r 0773f669c498 -r 4739af17e757 dom/webscheduling/WebTaskSchedulerMainThread.h
--- a/dom/webscheduling/WebTaskSchedulerMainThread.h	Thu Feb 27 16:17:36 2025 +0000
+++ b/dom/webscheduling/WebTaskSchedulerMainThread.h	Thu Feb 27 16:17:36 2025 +0000
@@ -32,6 +32,9 @@
   explicit WebTaskSchedulerMainThread(nsIGlobalObject* aParent)
       : WebTaskScheduler(aParent) {}
 
+  void IncreaseNumNormalOrHighPriorityQueuesHaveTaskScheduled() override;
+  void DecreaseNumNormalOrHighPriorityQueuesHaveTaskScheduled() override;
+
  private:
   nsresult SetTimeoutForDelayedTask(WebTask* aTask, uint64_t aDelay,
                                     EventQueuePriority aPriority) override;